bindings_generator/src/classes.rs

```diff
@@ -9,17 +9,23 @@ use quote::{format_ident, quote};
 
 use std::collections::HashMap;
 
-pub(crate) fn generate_class_struct(class: &GodotClass) -> TokenStream {
+pub(crate) fn generate_class_struct(class: &GodotClass, class_doc: TokenStream) -> TokenStream {
     let class_name = format_ident!("{}", &class.name);
 
     // dead_code: 'this' might not be read
+    // mod private: hide the type in the #module_name module, export it only in gdnative::api
     quote! {
-        #[allow(non_camel_case_types)]
-        #[derive(Debug)]
-        pub struct #class_name {
-            #[allow(dead_code)]
-            this: RawObject<Self>,
+        #[doc(hidden)]
+        pub(crate) mod private {
```

Comment:
`pub(crate)` modules probably don't need to be hidden?

Reply:
True.